File: android/src/main/java/kjd/reactnative/bluetooth/device/DeviceConnectionFactory.java
package kjd.reactnative.bluetooth.device;

import kjd.reactnative.RNBluetoothClassicModule;

/**
 * Provides the ability to create {@link DeviceConnection}.  The factory
 * is called by the {@link RNBluetoothClassicModule}
 * during the connection request process.
 *
 * @author kendavidson
 */
public interface DeviceConnectionFactory {

    /**
     * Creates a new {@link DeviceConnection}.
     *
     * @return
     */
    DeviceConnection create();
}

File: ios/conn/DeviceConnectionFactory.swift
//
//  DeviceConnectionFactory.swift
//  react-native-bluetooth-classic
//
//  Created by Ken Davidson on 2020-11-09.
//

import Foundation
import ExternalAccessory

/**
 * Provides a method for creating DeviceConnection(s).  The factories are used by the RNBluetoothClassic (module) during
 * the connection process.
 *
 * @author kendavidson
 */
protocol DeviceConnectionFactory {
    func create(accessory: EAAccessory, options: Dictionary<String,Any>) -> DeviceConnection
}
